### ISSUE TITLE
[strict-boolean-expressions] False positive for "true | undefined"

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#4103]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": "warn"
  }
}
```

```TS
interface Options {
  readonly enableSomething?: true;
}

function f(opts: Options): void {
  if (opts.enableSomething) console.log("Do something");
}
```

**Expected Result**

Property "enableSomething" has only one falsy value. Therefore there should be no error.

**Actual Result**

I get error:
"Unexpected nullable boolean value in conditional. Please handle the nullish case explicitly."

**Additional Info**

"allowNullableBoolean" option does not work here because it will disable errors for "boolean | undefined" which is not desired.

**Comment by JoshuaKGoldberg** [comment]
@ilyub does https://github.com/typescript-eslint/typescript-eslint/blob/076d65a/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md#allownullableboolean do what you're looking for?

**Comment by ilyub** [comment]
> 
> 
> @ilyub does https://github.com/typescript-eslint/typescript-eslint/blob/076d65a/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md#allownullableboolean do what you're looking for?

I think it does not.

This rule is supposed to show error when there are two or more Falsy values.

If you enable "allowNullableBoolean" option, it will ignore "boolean | undefined" which has two Falsy values: "false" and "undefined".

I DO want this rule to show error for "boolean | undefined", but I don't want it to show error for "true | undefined" because it has only one falsy value: "undefined".

**Comment by JoshuaKGoldberg** [comment]
Aha, thank you! That's a great point. Maybe `allowNullableTrue` as a separate option?

I know @bradzacher has talked with folks a lot about this in past issues such as #2894, just tagging to make sure I'm not missing something obvious 😄
